packages/runtime/lib/start.js

```diff
@@ -46,36 +58,15 @@ async function buildRuntime (configManager, env = process.env) {
   // The configManager cannot be transferred to the worker, so remove it.
   delete config.configManager
 
-  const worker = new Worker(kWorkerFile, {
-    /* c8 ignore next */
-    execArgv: config.hotReload ? kWorkerExecArgv : [],
-    transferList: config.loggingPort ? [config.loggingPort] : [],
-    workerData: { config, dirname, runtimeLogsDir },
-    env
-  })
+  let worker = startWorker({ config, dirname, runtimeLogsDir }, env)
 
   let managementApi = null
 
-  let exited = null
-  let isWorkerAlive = true
-  worker.on('exit', (code) => {
-    // TODO(mcollina): refactor to not set this here
-    process.exitCode = code
-    isWorkerAlive = false
-    configManager.fileWatcher?.stopWatching()
-    managementApi?.close()
-    if (typeof exited === 'function') {
-      exited()
-    }
-  })
-
-  worker.on('error', () => {
-    // If this is the only 'error' handler, then exit the process as the default
-    // behavior. If anything else is listening for errors, then don't exit.
-    if (worker.listenerCount('error') === 1) {
-      // The error is logged in the worker.
-      process.exit(1)
-    }
+  let exiting = false
+  closeWithGrace((event, cb) => {
+    exiting = true
+    worker.postMessage(event)
+    worker.once('exit', cb)
```

Comment:
Just a question. Can you move the exit logic here instead of setting an on `exit` listener to each worker?

```js
worker.once('exit, () => {
  configManager.fileWatcher?.stopWatching()
  managementApi?.close()
})
```

Reply:
done